src/citrine/resources/team.py

```diff
@@ -0,0 +1,291 @@
+from typing import Optional, Union, List
+from uuid import UUID
+
+from citrine._rest.collection import Collection
+from citrine._rest.resource import Resource, ResourceTypeEnum
+from citrine._serialization import properties
+from citrine._session import Session
+from citrine._utils.functions import format_escaped_url
+from citrine.resources.user import User
+
+WRITE = "WRITE"
+READ = "READ"
+SHARE = "SHARE"
+TEAM_ACTIONS = Union[WRITE, READ, SHARE]
+
+
+class TeamMember:
+    """A Member of a Team."""
+
+    def __init__(self,
+                 *,
+                 user: User,
+                 team: 'Team',  # noqa: F821
+                 actions: TEAM_ACTIONS):
+        self.user = user
+        self.team: 'Team' = team  # noqa: F821
+        self.actions: TEAM_ACTIONS = actions
+
+    def __str__(self):
+        return '<TeamMember {!r} can {!s} of {!r}>' \
+            .format(self.user.screen_name, self.actions, self.team.name)
+
+
+class Team(Resource['Team']):
+    """
+    A Citrine Team.
+
+    A Team is a collection of projects and resources.
+
+    Parameters
+    ----------
+    name: str
+        Name of the team.
+    description: str
+        Long-form description of the team.
+    session: Session, optional
+        The Citrine session used to connect to the database.
+
+    Attributes
+    ----------
+    uid: UUID
+        Unique uuid4 identifier of this team.
+    created_at: int
+        Time the team was created, in seconds since epoch.
+    name: str
+        Name of the Team
+    description: str
+        Description of the Team
+
+    """
+
+    _response_key = 'team'
+    _resource_type = ResourceTypeEnum.TEAM
+    _api_version = "v3"
+
+    name = properties.String('name')
+    description = properties.Optional(properties.String(), 'description')
+    uid = properties.Optional(properties.UUID(), 'id')
+    created_at = properties.Optional(properties.Datetime(), 'created_at')
+
+    def __init__(self,
+                 name: str,
+                 *,
+                 description: str = "",
+                 session: Optional[Session] = None):
+        self.name: str = name
+        self.description: str = description
+        self.session: Session = session
+
+    def __str__(self):
+        return '<Team {!r}>'.format(self.name)
+
+    def _path(self):
+        return format_escaped_url('/teams/{team_id}', team_id=self.uid)
+
+    def list_members(self) -> List[TeamMember]:
+        """
+        List all of the members in the current team.
+
+        Returns
+        -------
+        List[TeamMember]
+            The members of the current team
+
+        """
+        members = self.session.get_resource(self._path() + "/users",
+                                            version=self._api_version)["users"]
+        return [TeamMember(user=User.build(m), team=self, actions=m["actions"]) for m in members]
+
+    def remove_user(self, user_id: Union[str, UUID]) -> bool:
```

Comment:
Mostly because single user updates was the pattern in the previously used `projects` methods, and I stuck with it. We can add a `remove_users` method that can do many users at once (same for the other multi action methods)

Reply:
If single user updates is the existing pattern, then we should stick to it. Product can request multi-action methods later if there's a demand.